argo/providers/base/provider.go

```diff
@@ -15,8 +15,29 @@ type PipelineDefinition struct {
 	BeamArgs        map[string][]string `yaml:"beamArgs"`
 }
 
+type ExperimentDefinition struct {
+	Name        string `yaml:"name"`
+	Version     string `yaml:"version"`
+	Description string `yaml:"description"`
+}
+
+type RunConfigurationDefinition struct {
+	Name            string `yaml:"name"`
+	Version         string `yaml:"version"`
+	PipelineName    string `yaml:"pipelineName"`
+	PipelineVersion string `yaml:"pipelineVersion"`
+	ExperimentName  string `yaml:"experimentName"`
+	Schedule        string `yaml:"schedule"`
+}
+
 type Provider[Config any] interface {
 	CreatePipeline(providerConfig Config, pipelineDefinition PipelineDefinition, pipelineFile string, ctx context.Context) (string, error)
 	UpdatePipeline(providerConfig Config, pipelineDefinition PipelineDefinition, id string, pipelineFile string, ctx context.Context) (string, error)
-	DeletePipeline(providerConfig Config, pipelineDefinition PipelineDefinition, id string, ctx context.Context) error
+	DeletePipeline(providerConfig Config, id string, ctx context.Context) error
+
+	CreateRunConfiguration(providerConfig Config, runConfigurationDefinition RunConfigurationDefinition, ctx context.Context) (string, error)
```

Comment:
Wondering if the parameter name should. be something like `resourceDefinition` or `resource` instead of `runConfigurationDefinition` (for pipeline and experiments as well). After all we use a generic `id` instead of `runConfigurationId`.